src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Configuration/BaseProjectConfigurationDimensionProvider.cs

```diff
@@ -205,6 +213,27 @@ async Task UpdateUnderLockAsync(UnconfiguredProject project, Action<ProjectRootE
             }
         }
 
+        public virtual Task<IEnumerable<KeyValuePair<string, IEnumerable<string>>>> GetProjectConfigurationDimensionsAsync(Project project)
+        {
+            Requires.NotNull(project, nameof(project));
+
+            ImmutableArray<string> values = GetOrderedPropertyValues(project);
+            if (values.IsEmpty)
+            {
+                return Task.FromResult(Enumerable.Empty<KeyValuePair<string, IEnumerable<string>>>());
```

Comment:
We have a way to reuse cached empty enumerable tasks:

```suggestion
                return TaskResult.EmptyEnumerable<KeyValuePair<string, IEnumerable<string>>>();
```

Reply:
Done